src/server/pfs/server/driver.go

```diff
@@ -2186,27 +2242,54 @@ func (d *driver) getFile(ctx context.Context, file *pfs.File, offset int64, size
 	if err != nil {
 		return nil, err
 	}
-
 	paths, err := tree.Glob(file.Path)
 	if err != nil {
 		return nil, err
 	}
-	for path, node := range paths {
-		if node.FileNode == nil {
-			delete(paths, path)
+	var objects []*pfs.Object
+	var totalSize int64
+	foundDirectoryNode := false
+	for _, node := range paths {
+		if node.DirNode != nil {
+			foundDirectoryNode = true
+		}
+	}
+	if !foundDirectoryNode {
+		// e.g. this was a request for a single file e.g. /foo/bar.txt
+		dirNodePaths, err := tree.Glob(filepath.Dir(file.Path))
+		if err != nil {
+			return nil, err
+		}
+		if len(dirNodePaths) != 1 {
+			return nil, fmt.Errorf("invalid set of directory node paths - found %v expected 1 - this is probably a bug", len(dirNodePaths))
+		}
+		for key, node := range dirNodePaths {
+			paths[key] = node
```

Comment:
This code seems overly specific to reading everything from a single table, I don't think it's going to work for reading from a single datum in a table or reading from multiple tables. We should hammer out some more test cases for this and then I think it'll be clearer what this code should look like.

Reply:
Ah yea, very good point. We definitely need to handle those cases as well.